lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -2089,53 +2089,89 @@ diagnoseMatch(ModuleDecl *module, NormalProtocolConformance *conformance,
     break;
   }
 
-  case MatchKind::StaticNonStaticConflict:
-    // FIXME: Could emit a Fix-It here.
-    diags.diagnose(match.Witness, diag::protocol_witness_static_conflict,
-                   !req->isInstanceMember());
+  case MatchKind::StaticNonStaticConflict: {
+    auto witness = match.Witness;
+    auto diag = diags.diagnose(witness, diag::protocol_witness_static_conflict,
+                               !req->isInstanceMember());
+    if (req->isInstanceMember()) {
+      SourceLoc loc;
+      if (auto FD = dyn_cast<FuncDecl>(witness)) {
+        loc = FD->getStaticLoc();
+      } else if (auto VD = dyn_cast<VarDecl>(witness)) {
+        loc = VD->getParentPatternBinding()->getStaticLoc();
+      } else if (auto SD = dyn_cast<SubscriptDecl>(witness)) {
+        loc = SD->getStaticLoc();
+      } else {
+        llvm_unreachable("Unexpected witness");
```

Comment:
> rely on plain getStaticLoc() for anything else

What do you mean by that? Because I can't access `getStaticLoc()` until I cast it the witness to one of these specific decls.

> did we ever implement the thing where enum elements can satisfy requirements?

Nope, enum cases can't satisfy protocol requirements at the moment (see SR-3170).

Reply:
Oh, whoops. I thought `getStaticLoc` was on Decl or ValueDecl. Withdrawn.